roadmap/implementers-guide/src/node/approval/approval-voting.md

```diff
@@ -287,37 +288,78 @@ On receiving an `ApprovedAncestor(Hash, BlockNumber, response_channel)`:
 
 #### Determine Required Tranches
 
-This is pure logic is for inspecting an approval entry, containing the assignments received, the current time, requirements for approval, and the approval votes already received to determine how many of the delay tranches of the approval entry are relevant, as well as contextual information about what may be remaining to check on the candidate.
+This logic is for inspecting an approval entry that tracks the assignments received, along with information on which assignments have corresponding approval votes. Inspection also involves the current time and expected requirements and is used to help the higher-level code determine the following:
+  * Whether to broadcast the local assignment
+  * Whether to check that the candidate entry has been completely approved.
+  * If the candidate is waiting on approval, when to schedule the next wakeup of the `(candidate, block)` pair at a point where the state machine could be advanced.
+
+These routines are pure functions which only depend on the environmental state. The expectation is that this determination is re-run every time we attempt to update an approval entry: either when we trigger a wakeup to advance the state machine based on a no-show or our own broadcast, or when we receive further assignments or approvals from the network.
+
+Thus it may be that at some point in time, we consider that tranches 0..X is required to be considered, but as we receive more information, we might require fewer tranches. Or votes that we perceived to be missing and require replacement are filled in and change our view.
 
 Requires `(approval_entry, approvals_received, tranche_now, block_tick, no_show_duration, needed_approvals)`
 
 ```rust
 enum RequiredTranches {
   // All validators appear to be required, based on tranches already taken and remaining no-shows.
   All,
-  // More tranches required - We're awaiting more assignments. The given `DelayTranche` indicates the
-  // upper bound of tranches that should broadcast based on the last no-show.
-  Pending(DelayTranche),
+  // More tranches required - We're awaiting more assignments.
+  Pending {
+    /// The highest considered delay tranche when counting assignments.
+    considered: DelayTranche,
+    /// The tick at which the next no-show, of the assignments counted, would occur.
+    next_no_show: Option<Tick>,
+    /// The highest tranche to consider when looking to broadcast own assignment.
+    /// This should be considered along with the clock drift to avoid broadcasting
+    /// assignments that are before the local time.
+    maximum_broadcast: DelayTranche,
+    /// The clock drift, in ticks, to apply to the local clock when determining whether
+    /// to broadcast an assignment or when to schedule a wakeup. The local clock should be treated
+    /// as though it is `clock_drift` ticks earlier.
+    clock_drift: Tick,
```

Comment:
Should we say that `clock_drift` is caused by waiting for no-shows?

Reply:
I added a whole section on that earlier in the document so I decided to be lighter on detail here and focus on function rather than meaning